File: code-examples/scala/monitoring/cpm-subscribe.scala
import org.mongodb.scala._

// Replace the following with your MongoDB deployment's connection string
val uri: String = "mongodb+srv://<clusterUrl>/?replicaSet=rs&writeConcern=majority"

val client: MongoClient = MongoClient(uri)

// Replace <event name> with the name of the event you are subscribing to
val eventName: String  = "<event name>"

// Subscribe to the event
client.getDatabase("admin").watch().subscribe((event: ChangeEvent) => println("Received event: " + event), 
                                          (e: Throwable) => println("Error happened: " + e))

def run(): Unit = {
  try {
    // Establish and verify connection
    client.getDatabase("admin").runCommand(Document("ping" -> 1)).subscribe((result: Document) => println("Connected successfully!"),
                                                                        (e: Throwable) => println("Error happened during connection verification: " + e))
  } finally {
    // Ensures that the client will close when you finish/error
   client.close()
  }
}

try {
run()
} catch {
 case e: Exception => println("Error happened: " + e.getMessage)
}

File: code-examples/php/monitoring/cpm-subscribe.php
<?php
require 'vendor/autoload.php';

// Replace the following with your MongoDB deployment's connection string
$uri = "mongodb+srv://<clusterUrl>/?replicaSet=rs&writeConcern=majority";

$client = new MongoDB\Client($uri);

// Replace <event name> with the name of the event you are subscribing to
$eventName = "<event name>";

// Establish and verify connection
try {
    $ping = $client->admin->command(['ping' => 1]);
    echo "\nConnected successfully!\n";

    // Subscribe to the event
    $client->$eventName->insertOne(['event' => 'received']);

    // Fetch all records
    $allEvents = $client->$eventName->find();

    foreach($allEvents as $event) {
        var_dump($event);
    }

} catch(MongoDB\Driver\Exception\Exception $e) {
    echo $e->getMessage(), "\n";
} finally {
    // PHP MongoDB driver manages connections automatically, no need to close
}
?>
